perceval/backends/core/mbox.py

```diff
@@ -84,23 +83,115 @@ def fetch(self, from_date=DEFAULT_DATETIME):
 
         :returns: a generator of messages
         """
+        if not from_date:
+            from_date = DEFAULT_DATETIME
+
+        from_date = datetime_to_utc(from_date)
+
+        kwargs = {'from_date': from_date}
+        items = super().fetch("message", **kwargs)
+
+        return items
+
+    def fetch_items(self, **kwargs):
+        """Fetch the messages"""
+
+        from_date = kwargs['from_date']
+
         logger.info("Looking for messages from '%s' on '%s' since %s",
                     self.uri, self.dirpath, str(from_date))
 
-        mailing_list = MailingList(self.uri, self.dirpath)
+        self.client = MailingList(self.uri, self.dirpath)
```

Comment:
I wouldn't change the name of this variable. It's not exactly a client so I'd rather keep it as it is now.